### ISSUE TITLE
Extension overwrites workspace settings.json files.associations on load

### ISSUE THREAD

**Issue by JorisMolnar** [keesschollaart81/vscode-home-assistant#339]
## Expected behavior
When I change the `files.associations` setting in `.vscode/settings.json`, my setting does not get overwritten.

## Actual behavior
All changes to `files.associations` are overwritten when VS Code restarts and the Home Assistant extension loads.

## Context

Hi,

I'm running Home-Assistant in a docker container on a Ubuntu Server 20.04 host. I am editing my configuration using VS Code on a Windows machine with the 'Remote - SSH' extension to connect with the Ubuntu server.

My directory tree is as follows:
```
docker:
  - .vscode
    - settings.json
  - homeassistant
    - configuration.yaml
    - other-files.yaml
  - other-service
    - other-service-config.yaml
    - ...
```

During initialization, this extension creates the file `.vscode/settings.json`. This file contains the following:
```json
{
    "files.associations": {
        "*.yaml": "home-assistant"
    }
}
```

Because I also have other `.yaml` files in other subdirectories, I changed the file to:
```json
{
    "files.associations": {
        "**/homeassistant/**/*.yaml": "home-assistant"
    }
}
```

This works as expected. Only YAML files in the homeassistant directory get matched to the home-assistant language definition.

When I now restart VS Code, the `settings.json` file gets overwritten with the original configuration again. My setting `"**/homeassistant/**/*.yaml": "home-assistant"` gets changed into `"*.yaml": "home-assistant"`.

Extension runs in:
- [X] VS Code for Desktop 
- [ ] Hassio Add-on 
- [ ] Visual Studio Online
- [ ] Other: 

I'm running VS Code on:
- [X] Windows 
- [ ] Mac 
- [ ] Linux 

I'm accessing my files:
- [ ] From local disk 
- [X] Via remote SSH 
- [ ] Smb
- [ ] Other:

**Comment by frenck** [comment]
Fixed in #397

**Comment by crixionz** [comment]
Request to reopen. 
It still overwrites **all** file associations if the *.yaml->home-assistant rule is missing or manipulated. I think it is supposed to force this rule to be present, but not delete all other rules as a consequence.
